arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedColumnIterator.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.arrow.vectorized.parquet;
+
+import java.io.IOException;
+import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.iceberg.arrow.vectorized.NullabilityHolder;
+import org.apache.parquet.column.ColumnDescriptor;
+import org.apache.parquet.column.Dictionary;
+import org.apache.parquet.column.page.DataPage;
+import org.apache.parquet.column.page.DictionaryPage;
+import org.apache.parquet.column.page.PageReadStore;
+import org.apache.parquet.column.page.PageReader;
+import org.apache.parquet.io.ParquetDecodingException;
+
+/**
+ * Vectorized version of the ColumnIterator that reads column values in data pages of a column in a row group in a
+ * batched fashion.
+ */
+public class VectorizedColumnIterator {
+
+  private final ColumnDescriptor desc;
+  private final VectorizedPageIterator vectorizedPageIterator;
+
+  // state reset for each row group
+  private PageReader columnPageReader = null;
+  private long totalValuesCount = 0L;
+  private long valuesRead = 0L;
+  private long advanceNextPageCount = 0L;
+  private final int batchSize;
+
+  public VectorizedColumnIterator(ColumnDescriptor desc, String writerVersion, int batchSize) {
+    this.desc = desc;
+    this.batchSize = batchSize;
+    this.vectorizedPageIterator = new VectorizedPageIterator(desc, writerVersion, batchSize);
+  }
+
+  public Dictionary setRowGroupInfo(PageReadStore store, boolean allPagesDictEncoded) {
```

Comment:
Why does this need to be a `PageReadStore` instead of a `PageReader`? This implementation is nearly identical to `setPageSource` in the non-vectorized read path. I think we could refactor into a superclass and reuse `setPageSource` and `readDictionary` if we were to refactor into a common superclass. Probably don't need to do that in this PR, but it would be nice to avoid unnecessary drift between the two read implementations.

Reply:
We should do this in a follow-up.